### ISSUE TITLE
Error 400 Bad Request from Google API

### ISSUE THREAD

**Issue by moults31** [SpotlightKid/mrequests#4]
Hi, nice module! I'm running into a bit of a specific issue here where a seemingly simple `GET` request succeeds under `requests` fails under `mrequests`.

### Error signature:
```
access_token = make_jwt()   # make_jwt implementation is copy pasted from one of the tutorial links below
headers = {'authorization': f'Bearer {access_token}',
           'Content-Type': 'application/json'}
# This url is for a public tutorial sheet, use it exactly as-is
url = 'https://sheets.googleapis.com/v4/spreadsheets/1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms/values/Class Data!A:F'
r = requests.get(url, headers=headers)
```
Returns 200 when using `requests` but returns the following error when using `mrequests`:
```
<title>Error 400 (Bad Request)!!1</title>
...
  <p><b>400.</b> <ins>That’s an error.</ins>
  <p>Your client has issued a malformed or illegal request.  <ins>That’s all we know.</ins>
```

### Steps to reproduce:
I followed [this Medium article](https://owaisqureshi.medium.com/access-google-sheets-api-in-python-using-service-account-3a0c6d89d5fc) to set up a Google app that can access the aforementioned URL using the recommended Google python modules.
Then I followed [this Stackoverflow answer](https://stackoverflow.com/a/59311672) to convert that script into one that uses `requests`. This whole process only takes about half an hour or less to get up and running.
Then I imported my `mrequests` clone in place of `requests` (my goal is to run this on a Pi Pico W under micropython) and encountered the error signature above.

### Notes:
- I encounter the same error response under python3, micropython Unix port, and micropython on the Pi Pico W.
- I encounter the same error response when using `urequests`
- Note that the make_jwt method uses RS256 which I haven't found a micropython implementation for, so that part needs to be done on the big Python. However the token that make_jwt returns can be run on any machine until it expires, so in theory it could be transferred to a micropython machine as a string.
- Here's the headers section of the 200 response in case anything here is useful:
```
{'Content-Type': 'application/json; charset=UTF-8', 'Vary': 'Origin, X-Origin, Referer', 'Content-Encoding': 'gzip', 'Date': 'Mon, 10 Oct 2022 02:07:58 GMT', 'Server': 'ESF', 'Cache-Control': 'private', 'X-XSS-Protection': '0', 'X-Frame-Options': 'SAMEORIGIN', 'X-Content-Type-Options': 'nosniff', 'Alt-Svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"', 'Transfer-Encoding': 'chunked'}
UTF-8
```

I'll keep digging into this and try to see if I can find where this particular request starts to differ between `requests` and `mrequests` and will update here if I find anything. @SpotlightKid if you could find some time to try to reproduce this that would be greatly appreciated!! Any questions, fire away.

Zac

**Comment by moults31** [comment]
Big update: I've discovered that the issue only affects Google Sheets with a space character in their name. So the sheet referenced in the linked tutorials named 'Class Data' is affected but a custom sheet I made named 'ClassData' is not. For sheets without spaces in their names, I get a 200 Ok response under all of Python3 mrequests, micropython mrequests, and micropython urequests.

So, I am effectively unblocked since I can just make sure my sheets never have spaces in their names.

I'll let @SpotlightKid decide whether to keep this issue open or not. If anyone is interested in pursuing it, I'd say the prime suspect is some kind of encoding issue for that space character in the sheet name, which urrlib3/python3-requests is somehow not affected by.
